internal/test/e2e/otelglobal/integration_test.go

```diff
@@ -7,9 +7,107 @@ package autosdk
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/goleak"
+
 	"go.opentelemetry.io/auto/internal/test/e2e"
 )
 
+// scopeName defines the instrumentation scope name used in the trace.
+const scopeName = "trace-example"
+
 func TestIntegration(t *testing.T) {
-	e2e.TestIntegration(t, "./cmd", ".")
+	if testing.Short() {
+		t.Skip("skipping long-running integration test in short mode.")
+	}
+
+	defer goleak.VerifyNone(t)
+
+	traces := e2e.RunInstrumentation(t, "./cmd")
+	scopes := e2e.ScopeSpansByName(traces, scopeName)
+	require.NotEmpty(t, scopes)
+
+	t.Run("ResourceAttribute/ServiceName", func(t *testing.T) {
+		val, err := e2e.ResourceAttribute(traces, "service.name")
+		require.NoError(t, err)
+		assert.Equal(t, "sample-app", val.AsString())
+	})
+
+	t.Run("Scope", func(t *testing.T) {
+		assert.Equal(t, scopeName, scopes[0].Scope().Name(), "scope name")
+		assert.Equal(t, "v1.23.42", scopes[0].Scope().Version(), "scope version")
+	})
+
+	t.Run("SchemaURL", func(t *testing.T) {
+		assert.Equal(t, "https://some_schema", scopes[0].SchemaUrl())
+	})
+
+	parentSpan, err := e2e.SpanByName(scopes, "parent")
+	require.NoError(t, err)
+	t.Run("ParentSpan", func(t *testing.T) {
+		e2e.AssertTraceID(t, parentSpan.TraceID(), "trace ID")
+		e2e.AssertSpanID(t, parentSpan.SpanID(), "span ID")
+
+		assert.Equal(t, ptrace.SpanKindServer, parentSpan.Kind(), "span kind")
+
+		status := parentSpan.Status()
+		assert.Equal(t, ptrace.StatusCodeOk, status.Code(), "status code")
+
+		attrs := e2e.AttributesMap(parentSpan.Attributes())
+		assert.Equal(t, int64(42), attrs["int_key"].Int(), "int_key")
```

Comment:
Won't this panic if `attrs["int_key"]` does not exist?

Reply:
It will, but this is also a test. The panic is validating the existence, just will poor representations of the source.

I'm fine addressing this, but would like to do so holistically for all end-to-end test once we get these migrated.

Reply:
https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2229